anvil-python/anvil/commands/juju.py

```diff
@@ -54,3 +63,96 @@ def run(self, status: Status | None) -> Result:
                 message="Could not find public ssh key (~/.ssh/id_rsa.pub)",
             )
         return Result(ResultType.COMPLETED)
+
+
+class ScaleJujuStep(BaseStep, JujuStepHelper):
+    """Enable Juju HA."""
+
+    def __init__(
+        self,
+        jhelper: JujuHelper,
+        model: str,
+    ):
+        super().__init__("Juju HA", "Enable Juju High Availability")
+
+        self.jhelper = jhelper
+        self.model = model
+
+        self.controller_machines: set[str] = set()
+        self.machines: set[str] = set()
+
+    def run(self, status: Status | None = None) -> Result:
+        """Run the step to completion."""
+
+        available_machines = list(self.machines ^ self.controller_machines)
+        n_machines_to_join = min(
+            len(available_machines),
+            MAX_JUJU_CONTROLLERS - len(self.controller_machines),
+        )
+
+        cmd = [
```

Comment:
This command should be wrapped with try/expect since it can fail and we need to know why. This is a failure than can happen because of the linked bug:

```bash
ubuntu@maas-1:~$ juju enable-ha -n 3 --to 3
ERROR juju-ha-space is not set and a unique usable address was not found for machines: 0
```
With the current code, we fail in an unexpected way:

```bash
EBUG    Command '['/snap/maas-anvil/x1/juju/bin/juju', 'enable-ha', '-n', '3', '--to', '3']' returned non-zero exit status 1.                                                        utils.py:38
                    Traceback (most recent call last):                                                                                                                                                      
                      File "/snap/maas-anvil/x1/lib/python3.10/site-packages/anvil/utils.py", line 32, in __call__                                                                                          
                        return self.main(*args, **kwargs)                                                                                                                                                   
                      File "/snap/maas-anvil/x1/lib/python3.10/site-packages/click/core.py", line 1078, in main                                                                                             
                        rv = self.invoke(ctx)                                                                                                                                                               
                      File "/snap/maas-anvil/x1/lib/python3.10/site-packages/click/core.py", line 1688, in invoke                                                                                           
                        return _process_result(sub_ctx.command.invoke(sub_ctx))                                                                                                                             
                      File "/snap/maas-anvil/x1/lib/python3.10/site-packages/click/core.py", line 1688, in invoke                                                                                           
                        return _process_result(sub_ctx.command.invoke(sub_ctx))                                                                                                                             
                      File "/snap/maas-anvil/x1/lib/python3.10/site-packages/click/core.py", line 1434, in invoke                                                                                           
                        return ctx.invoke(self.callback, **ctx.params)                                                                                                                                      
                      File "/snap/maas-anvil/x1/lib/python3.10/site-packages/click/core.py", line 783, in invoke                                                                                            
                        return __callback(*args, **kwargs)                                                                                                                                                  
                      File "/snap/maas-anvil/x1/lib/python3.10/site-packages/click/decorators.py", line 33, in new_func                                                                                     
                        return f(get_current_context(), *args, **kwargs)                                                                                                                                    
                      File "/snap/maas-anvil/x1/lib/python3.10/site-packages/anvil/provider/local/commands.py", line 599, in remove                                                                         
                        run_plan(plan, console)                                                                                                                                                             
                      File "/snap/maas-anvil/x1/lib/python3.10/site-packages/sunbeam/jobs/common.py", line 277, in run_plan                                                                                 
                        result = step.run(status)                                                                                                                                                           
                      File "/snap/maas-anvil/x1/lib/python3.10/site-packages/anvil/commands/juju.py", line 186, in run                                                                                      
                        process = subprocess.run(                                                                                                                                                           
                      File "/usr/lib/python3.10/subprocess.py", line 526, in run                                                                                                                            
                        raise CalledProcessError(retcode, process.args,                                                                                                                                     
                    subprocess.CalledProcessError: Command '['/snap/maas-anvil/x1/juju/bin/juju', 'enable-ha', '-n', '3', '--to', '3']' returned non-zero exit status 1.                                    
           WARNING  An unexpected error has occurred. Please run 'maas-anvil inspect' to generate an inspection report.                                                                          utils.py:43
           ERROR    Error: Command '['/snap/maas-anvil/x1/juju/bin/juju', 'enable-ha', '-n', '3', '--to', '3']' returned non-zero exit status 1.                                                 utils.py:44
           ERROR    Task was destroyed but it is pending!                                                                                                                                base_events.py:1758
                    task: <Task pending name='Task-59' coro=<Connection._pinger.<locals>._do_ping() done, defined at                                                                                        
                    /snap/maas-anvil/x1/lib/python3.10/site-packages/juju/client/connection.py:599> wait_for=<Future cancelled>                                                                             
                    cb=[create_task_with_handler.<locals>._task_result_exp_handler(task_name='tmp', logger=<Logger juju....ction (ERROR)>)() at                                                             
                    /snap/maas-anvil/x1/lib/python3.10/site-packages/juju/jasyncio.py:39]>
```